public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -236,4 +247,112 @@ public ConceptIds classifyConceptIds(Set<Long> conceptIds) {
         }
         return new ConceptIds(standardConceptIds.build(), sourceConceptIds.build());
     }
+
+    public List<Concept> getStandardConcepts(Long conceptId) {
+        return conceptDao.findStandardConcepts(conceptId).stream()
+                .map(conceptMapper::dbModelToClient)
+                .collect(Collectors.toList());
+    }
+
+    public List<Concept> getSourceConcepts(Long conceptId, Integer count) {
+        return conceptDao.findSourceConcepts(conceptId, count).stream()
+                .map(conceptMapper::dbModelToClient)
+                .collect(Collectors.toList());
+    }
+
+    public List<Long> getDrugIngredientsByBrand(String query) {
+        return conceptDao.findDrugIngredientsByBrand(query);
+    }
+
+    public ConceptListResponse getConcepts(SearchConceptsRequest searchConceptsRequest) {
+        Integer maxResults = searchConceptsRequest.getMaxResults();
+        if(maxResults == null || maxResults == 0){
+            maxResults = Integer.MAX_VALUE;
+        }
+        List<Long> drugConcepts = new ArrayList<>();
+
+        if(searchConceptsRequest.getDomain() != null && searchConceptsRequest.getDomain().equals(Domain.DRUG) && searchConceptsRequest.getQuery() != null && !searchConceptsRequest.getQuery().isEmpty()) {
+            List<Long> drugMatchedConcepts = new ArrayList<>();
+            drugMatchedConcepts = getDrugIngredientsByBrand(searchConceptsRequest.getQuery());
+            if(drugMatchedConcepts.size() > 0) {
+                drugConcepts = drugMatchedConcepts;
+            }
+        }
+
+        Integer minCount = searchConceptsRequest.getMinCount();
+        if(minCount == null){
+            minCount = 1;
+        }
+
+        StandardConceptFilter standardConceptFilter = searchConceptsRequest.getStandardConceptFilter();
+
+
+        if(searchConceptsRequest.getQuery() == null || searchConceptsRequest.getQuery().isEmpty()){
```

Comment:
`org.apache.commons.lang3.StringUtils.isEmpty()` checks for empty ("") or null
```suggestion
       if(StringUtils.isEmpty(searchConceptsRequest.getQuery())){
```

Reply:
Changed this.